compile error in GPU

```
libpaddle_fluid.a(pool_op.cu.cc.o):/usr/include/c++/5/bits/basic_ios.h:372: first defined here
libpaddle_fluid.a(pool_op.cu.cc.o): In function `paddle::operators::PoolGradKernel<paddle::platform::CUDADeviceContext, double>::Compute(paddle::framework::ExecutionContext const&) const':
pool_op.cu.cc:(.text._ZNK6paddle9operators14PoolGradKernelINS_8platform17CUDADeviceContextEdE7ComputeERKNS_9framework16ExecutionContextE[_ZNK6paddle9operators14PoolGradKernelINS_8platform17CUDADeviceContextEdE7ComputeERKNS_9framework16ExecutionContextE]+0x85a): undefined reference to `Eigen::internal::TensorExecutor<Eigen::TensorAssignOp<Eigen::TensorMap<Eigen::Tensor<double, 1, 1, long>, 0, Eigen::MakePointer>, Eigen::TensorCwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::TensorMap<Eigen::Tensor<double, 1, 1, long>, 0, Eigen::MakePointer> const> const> const, Eigen::GpuDevice, false>::run(Eigen::TensorAssignOp<Eigen::TensorMap<Eigen::Tensor<double, 1, 1, long>, 0, Eigen::MakePointer>, Eigen::TensorCwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::TensorMap<Eigen::Tensor<double, 1, 1, long>, 0, Eigen::MakePointer> const> const> const&, Eigen::GpuDevice const&)'
libpaddle_fluid.a(pool_op.cu.cc.o): In function `paddle::operators::PoolGradKernel<paddle::platform::CUDADeviceContext, float>::Compute(paddle::framework::ExecutionContext const&) const':
pool_op.cu.cc:(.text._ZNK6paddle9operators14PoolGradKernelINS_8platform17CUDADeviceContextEfE7ComputeERKNS_9framework16ExecutionContextE[_ZNK6paddle9operators14PoolGradKernelINS_8platform17CUDADeviceContextEfE7ComputeERKNS_9framework16ExecutionContextE]+0x839): undefined reference to `Eigen::internal::TensorExecutor<Eigen::TensorAssignOp<Eigen::TensorMap<Eigen::Tensor<float, 1, 1, long>, 0, Eigen::MakePointer>, Eigen::TensorCwiseNullaryOp<Eigen::internal::scalar_constant_op<float>, Eigen::TensorMap<Eigen::Tensor<float, 1, 1, long>, 0, Eigen::MakePointer> const> const> const, Eigen::GpuDevice, false>::run(Eigen::TensorAssignOp<Eigen::TensorMap<Eigen::Tensor<float, 1, 1, long>, 0, Eigen::MakePointer>, Eigen::TensorCwiseNullaryOp<Eigen::internal::scalar_constant_op<float>, Eigen::TensorMap<Eigen::Tensor<float, 1, 1, long>, 0, Eigen::MakePointer> const> const> const&, Eigen::GpuDevice const&)'
```

We can not use  eigen in pool_op.h, it leads to compile error in pool_op.cu.cc

--- a/paddle/operators/pool_op.h
+++ b/paddle/operators/pool_op.h
@@ -139,10 +139,8 @@ class PoolGradKernel : public framework::OpKernel<T> {
     auto& dev_ctx = context.template device_context<DeviceContext>();
     if (in_x_grad) {
       in_x_grad->mutable_data<T>(context.GetPlace());
-      auto temp = framework::EigenVector<T>::Flatten(*in_x_grad);
-      temp.device(
-          *context.template device_context<DeviceContext>().eigen_device()) =
-          temp.constant(static_cast<T>(0));
+      paddle::operators::math::SetConstant<DeviceContext, T> set_constant;
+      set_constant(dev_ctx, in_x_grad, 0.0);
 
       switch (ksize.size()) {
         case 2: {

